### PR TITLE
Update regex for telegram

### LINKED ISSUE #146: Regex for Telegram Issue not working for those starting with invalid examples
Regex for Telegram Issue not working for those starting with invalid examples, update the UG rules for this

### PR DESCRIPTION
**Pull request by Vanessamae23**
## Description

Telegram handle cannot end with an underscore and must start with a letter fixed with updated UG parameter

## Context

Closes #146 

## Checklist

- [ ] I have self-reviewed my changes.
- [ ] I have added JavaDocs to all relevant methods.
- [ ] I have updated the documentation: 
  - User Guide Table of Contents, Description and Summary.
  - Developer Guide, if applicable.
- [ ] I have ran `./gradlew test` or `./gradlew check` and all checks pass.
- [ ] I have updated my project portfolio with what I have contributed.

### CODE REVIEW

File: docs/UserGuide.md
Comment (proto-aiken-13) [inline docs/UserGuide.md:1]: No issues.

File: src/test/java/seedu/address/model/person/TelegramHandleTest.java
Comment (proto-aiken-13) [inline src/test/java/seedu/address/model/person/TelegramHandleTest.java:1]: LGTM

**Review by proto-aiken-13 — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/person/TelegramHandle.java [hunk 1]
```diff
@@ -11,13 +11,14 @@ public class TelegramHandle {
...
     public static final TelegramHandle NULL_TELEGRAM_HANDLE = null;
     public static final String MESSAGE_CONSTRAINTS = "Telegram Handle must "
-            + "use a-z, 0-9 and underscores. Minimum length is 5 characters";
+            + "use a-z, 0-9 and underscores. Minimum length is 5 characters. It must start with a letter and cannot "
+            + "end with an underscore";
 
     /*
      * The telegram handle uses a-z, 0-9 and underscores.
      * Minimum length is 5 characters.
      */
```
Comment (et-irl) [inline src/main/java/seedu/address/model/person/TelegramHandle.java:20]: Did you forget to update the comment explaining the regex to be similar to `MESSAGE_CONSTRAINTS`?

File: src/test/java/seedu/address/model/person/TelegramHandleTest.java [hunk 2]
```diff
@@ -21,18 +21,18 @@ public void constructor_invalidTelegramHandle_throwsIllegalArgumentException() {
...
         assertFalse(TelegramHandle.isValidTelegramHandle("")); // empty string
         assertFalse(TelegramHandle.isValidTelegramHandle(" ")); // spaces only
         assertFalse(TelegramHandle.isValidTelegramHandle("1234")); // less than 5
         assertFalse(TelegramHandle.isValidTelegramHandle("%&@#&@")); // contains invalid symbols
 
-        // valid addresses
+        // valid telegrams
         assertTrue(TelegramHandle.isValidTelegramHandle("mamamia"));
-        assertTrue(TelegramHandle.isValidTelegramHandle("12345")); // 5 digits
+        assertTrue(TelegramHandle.isValidTelegramHandle("lol_lol")); // 5 digits
```
Comment (et-irl) [inline src/test/java/seedu/address/model/person/TelegramHandleTest.java:35]: Did you forget to update the comment explaining this test?

File: src/main/java/seedu/address/model/person/TelegramHandle.java [hunk 3]
```diff
@@ -11,13 +11,14 @@ public class TelegramHandle {
...
-    public static final String VALIDATION_REGEX = "^[a-zA-Z0-9_]{5,32}$";
+    public static final String VALIDATION_REGEX = "^[a-zA-Z][a-zA-Z0-9_]{3,28}[a-zA-Z0-9]$";
```
Comment (et-irl) [inline src/main/java/seedu/address/model/person/TelegramHandle.java:21]: Is there an explanation for the upper bound on 28? Is the maximum handle length 30? Explain in the comment above for the regex.

**Review by et-irl — CHANGES_REQUESTED** [review]
Some documentation errors, otherwise, nice job.

File: src/test/java/seedu/address/model/person/TelegramHandleTest.java [hunk 4]
```diff
@@ -21,18 +21,18 @@ public void constructor_invalidTelegramHandle_throwsIllegalArgumentException() {
...
         assertTrue(TelegramHandle.isValidTelegramHandle("longtelegramhandle")); // long telegram handle
```
Comment (et-irl) [inline src/test/java/seedu/address/model/person/TelegramHandleTest.java:36]: This comment does not look useful, is this in the list of bad software engineering practices?

Comment (Vanessamae23) [reply src/test/java/seedu/address/model/person/TelegramHandleTest.java:36]: updated thanks

File: src/main/java/seedu/address/model/person/TelegramHandle.java (hunk 3 above)
Comment (Vanessamae23) [reply src/main/java/seedu/address/model/person/TelegramHandle.java:22]: Updated thanks

File: src/test/java/seedu/address/model/person/TelegramHandleTest.java (hunk 2 above)
Comment (Vanessamae23) [reply src/test/java/seedu/address/model/person/TelegramHandleTest.java]: updated thanks

File: src/main/java/seedu/address/model/person/TelegramHandle.java (hunk 1 above)
Comment (Vanessamae23) [reply src/main/java/seedu/address/model/person/TelegramHandle.java]: updated thanks

File: src/test/java/seedu/address/model/person/TelegramHandleTest.java (hunk 4 above)
Comment (Vanessamae23) [reply src/test/java/seedu/address/model/person/TelegramHandleTest.java]: updated thanks

File: src/main/java/seedu/address/model/person/TelegramHandle.java (hunk 3 above)
Comment (Vanessamae23) [reply src/main/java/seedu/address/model/person/TelegramHandle.java]: max is 32 characters, my bad for not specifying

File: src/test/java/seedu/address/model/person/TelegramHandleTest.java [hunk 5]
```diff
@@ -29,11 +29,13 @@ public void isValidTelegramHandle() {
...
+        assertFalse(TelegramHandle.isValidTelegramHandle("ggggggggggggggggggggggggggggggggg")); // too long
+        assertFalse(TelegramHandle.isValidTelegramHandle("four")); // too short
 
         // valid telegrams
         assertTrue(TelegramHandle.isValidTelegramHandle("mamamia"));
-        assertTrue(TelegramHandle.isValidTelegramHandle("lol_lol")); // 5 digits
-        assertTrue(TelegramHandle.isValidTelegramHandle("longtelegramhandle")); // long telegram handle
+        assertTrue(TelegramHandle.isValidTelegramHandle("lol_lol")); // underscore
+        assertTrue(TelegramHandle.isValidTelegramHandle("v32ggggggggggggggggggggggggggggg")); // valid long
     }
```
Comment (et-irl) [inline src/test/java/seedu/address/model/person/TelegramHandleTest.java:39]: Love the test case around boundaries!

**Review by et-irl — APPROVED** [review]
Looks great! 🚀 🚀

**Pull request merged**